operators/pkg/tenant-controller/sandbox.go

```diff
@@ -0,0 +1,143 @@
+	tenant.Status.SandboxNamespace.Name = sandboxnsName
+
+	// Enforce role binding precence
+	roleBind := rbacv1.RoleBinding{ObjectMeta: metav1.ObjectMeta{Name: "sandbox-editor", Namespace: sandboxnsName}}
+	res, err = ctrl.CreateOrUpdate(ctx, r.Client, &roleBind, func() error {
+		roleBind.SetLabels(forge.SandboxObjectLabels(roleBind.GetLabels(), tenant.Name))
+		roleBind.RoleRef = rbacv1.RoleRef{Kind: "ClusterRole", Name: "edit", APIGroup: rbacv1.GroupName}
+		roleBind.Subjects = []rbacv1.Subject{{Kind: rbacv1.UserKind, Name: tenant.Name, APIGroup: rbacv1.GroupName}}
+		return ctrl.SetControllerReference(tenant, &roleBind, r.Scheme)
+	})
+	if err != nil {
+		log.Error(err, "failed to enforce resource", "role binding", klog.KObj(&roleBind))
+		return err
+	}
+	log.V(utils.FromResult(res)).Info("sandbox role binding correctly enforced", "role binding", klog.KObj(&roleBind), "result", res)
+
+	// Enforce resource quota presence
+	resourceQuota := v1.ResourceQuota{
+		ObjectMeta: metav1.ObjectMeta{Name: "sandbox-resource-quota", Namespace: sandboxnsName},
+	}
+	res, err = ctrl.CreateOrUpdate(ctx, r.Client, &resourceQuota, func() error {
+		resourceQuota.SetLabels(forge.SandboxObjectLabels(resourceQuota.GetLabels(), tenant.Name))
+		resourceQuota.Spec.Hard = forge.SandboxResourceQuotaSpec()
+		return ctrl.SetControllerReference(tenant, &resourceQuota, r.Scheme)
+	})
+	if err != nil {
+		log.Error(err, "failed to enforce resource", "resource quota", klog.KObj(&resourceQuota))
+		return err
+	}
+	log.V(utils.FromResult(res)).Info("sandbox resource quota correctly enforced", "resource quota", klog.KObj(&resourceQuota), "result", res)
+	tenant.Status.SandboxNamespace.Created = true
+
+	return err
+}
+
+// enforceInstanceExpositionAbsence ensures the sandbox's absence.
+func (r *TenantReconciler) enforceSandboxResourcesAbsence(ctx context.Context) error {
+	tenant := clctx.TenantFrom(ctx)
+	sandboxnsName := fmt.Sprintf("sandbox-%s", strings.ReplaceAll(tenant.Name, ".", "-"))
+
+	// Enforce namespace absence
+	namespace := v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: sandboxnsName}}
+	if err := r.enforceObjectAbsence(ctx, &namespace, "namespace"); err != nil {
+		return err
+	}
+
+	tenant.Status.SandboxNamespace.Created = false
+	tenant.Status.SandboxNamespace.Name = ""
+
+	return nil
+}
+
+// enforceObjectAbsence deletes a Kubernetes object and prints the appropriate log messages, without failing if it does not exist.
+func (r *TenantReconciler) enforceObjectAbsence(ctx context.Context, obj client.Object, kind string) error {
+	if err := r.Delete(ctx, obj); err != nil {
+		if !kerrors.IsNotFound(err) {
+			ctrl.LoggerFrom(ctx).Error(err, "failed to delete object", kind, klog.KObj(obj))
+			return err
+		}
+		ctrl.LoggerFrom(ctx).V(utils.LogDebugLevel).Info("the object was already removed", kind, klog.KObj(obj))
+	} else {
+		ctrl.LoggerFrom(ctx).V(utils.LogInfoLevel).Info("object correctly removed", kind, klog.KObj(obj))
+	}
+
+	return nil
+}
```

Comment:
This function should be appropriately generalized to prevent code duplication with the one in `exposition.go`